src/libraries/Common/src/Interop/Linux/OpenLdap/Interop.Ldap.cs

```diff
@@ -77,8 +79,8 @@ static Ldap()
         [DllImport(Libraries.OpenLdap, EntryPoint = "ldap_initialize", CharSet = CharSet.Ansi, SetLastError = true)]
         public static extern int ldap_initialize(out IntPtr ld, string hostname);
 
-        [DllImport(Libraries.OpenLdap, EntryPoint = "ldap_init", CharSet = CharSet.Ansi, SetLastError = true)]
-        public static extern IntPtr ldap_init(string hostName, int portNumber);
+        [DllImport(Libraries.OpenLdap, EntryPoint = "ldap_create", CharSet = CharSet.Ansi, SetLastError = true)]
```

Comment:
Can you share the difference between ldap_init and ldap_create?

Reply:
`ldap_init` and `ldap_initialize` both call `ldap_create` internally. `ldap_create` does the heavy lifting of allocating memory and instantiating global options and whatnot: `ldap_init` and `ldap_initialize` are just wrappers to easily specify the server using a hostname and port or an LDAP URI, respectively.

`ldap_init` has been deprecated by the OpenLDAP developers for a few reasons, one of which is the inability to specify the protocol scheme using `ldap_init`. Lbraries that are not compiled with `#define LDAP_DEPRECATED 1` will not include `ldap_init`. The OpenLDAP developers [recommend](https://git.openldap.org/openldap/openldap/-/blob/OPENLDAP_REL_ENG_2_4_58/include/ldap.h#L1511-1515) using either `ldap_create` or `ldap_initialize` to replace `ldap_init`.

FWIW, we use `ldap_create` instead of  `ldap_initialize` when setting up an `LdapConnection` since we don't know the value of the `SessionOptions.SecureSocketLayer` property when we initialize the OpenLDAP handle.

Reply:
Makes sense, thanks so much for the explanation. I agree with you that `ldap_init` is deprecated and we shouldn't be using it any longer. The [docs from openldap](https://www.openldap.org/software//man.cgi?query=ldap_init&sektion=3&apropos=0&manpath=OpenLDAP+2.4-Release) seem to only reference `ldap_initialize` though, so I wonder if we should be using that instead of `ldap_create`. What are your thoughts?

Reply:
I don't mind either way; it winds up doing the same thing. 

What I was trying by using `ldap_create` over `ldap_initialize` to do was avoid  setting the `LDAP_OPT_URI` option twice (once during initialization, and again after the `SessionOptions` is populated). But after reading through the code for `ldap_initialize` again I've discovered we can accomplish that by passing a `null` value as the URI. That would eliminate the need to call `ldap_create`. I can do that in the morning if that's your preference.

> The docs from openldap seem to only reference `ldap_initialize`

(Reading through the mailing list, I think that the developers themselves admit that there's some stuff missing from the manpages. I don't know if `ldap_create`'s absence in the manpage is intentional or not.)

Reply:
(even when I don’t really think that any of the two will ever be changed, but might as well be more careful)